pkg/cache/cache_test.go

```diff
@@ -0,0 +1,74 @@
+package cache_test
```

Comment:
not enough tests. You need to check that your cache works well when being called concurrently. So call methods from different goroutines.